pytorch_lightning/trainer/connectors/logger_connector/result.py

```diff
@@ -289,9 +289,12 @@ class ResultMetricCollection(dict):
     with the same metadata.
     """
 
-    def __init__(self, *args, metadata: Optional[_Metadata] = None) -> None:
+    def __init__(self, *args) -> None:
         super().__init__(*args)
-        self.meta = metadata
+
+    @property
+    def meta(self) -> _Metadata:
+        return list(self.values())[0].meta
```

Comment:
do we need a sanity check that meta from all values is the same?

Reply:
No, the meta is automaticly set by us